### PR TITLE
feat(pie-monorepo): DSW-000 upgrade to lit@3

### PR DESCRIPTION
**Pull request by raoufswe**
## Describe your changes (can list changeset entries if preferable)
- Updated lit to the latest version `3.1.0`
- Migrated the react wrapped component to use `@lit/react`
- Ensured that all apps are compatible and work as expected 
- The packages were released with the `next` tag and have been tested in [pie-aperture](https://github.com/justeattakeaway/pie-aperture/pull/30)
- Updated several packages on the way such as typescript and other testing packages.

## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: packages/tools/pie-icons-react/package.json [hunk 1]
```diff
@@ -50,17 +50,20 @@
...
+    "react-test-renderer": "18.0.0",
     "rollup": "3.20.2",
     "rollup-plugin-typescript2": "0.34.1",
-    "typescript": "5.1.3"
+    "typescript": "5.3.2"
   },
   "dependencies": {
-    "@babel/preset-react": "7.18.6",
-    "react-test-renderer": "18.0.0"
+    "@justeattakeaway/pie-icons": "4.9.2",
```
Comment (jamieomaguire) [inline packages/tools/pie-icons-react/package.json:63]: Was this necessary for the Lit upgrade? I think we should be cautious about adding additional changes like this to the PR. I would prefer we do these kinds of things separately to minimise risk

Comment (raoufswe) [reply packages/tools/pie-icons-react/package.json:63]: yes, it was unfortunately, I will share the Slack thread with you for this in a bit.

**Review by siggerzz — DISMISSED** [review]

File: configs/pie-components-config/vite.config.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ const sharedConfig = ({ build = {}, plugins = [], ...rest }) => defineConfig({
         },
         rollupOptions: {
             external: (id) => {
```
Comment (fernandofranca) [inline configs/pie-components-config/vite.config.js:24]: Having a comment explaining why we use those `external` configs can be quite handy in the future.

**Review by siggerzz — APPROVED** [review]

File: configs/pie-components-config/vite.config.js (hunk 2 above)
Comment (raoufswe) [reply configs/pie-components-config/vite.config.js:24]: good idea! is it okay if i address that in a follow up PR?

Comment (fernandofranca) [reply configs/pie-components-config/vite.config.js:24]: For sure!

**Review by fernandofranca — APPROVED** [review]

**Pull request merged**